module/purpose/Soccer/src/Soccer.hpp

```diff
@@ -68,6 +76,36 @@ namespace module::purpose {
             Position position{};
         } cfg;
 
+        struct RobotInfo {
+            uint8_t robot_id;
+            std::chrono::steady_clock::time_point last_heard_from;
+
+            bool operator<(const RobotInfo& other) const {
+                return robot_id < other.robot_id;
+            }
+        };
+
+        /// @brief Add and update active robots
+        void manage_active_robots(const RoboCup& robocup);
+
+        /// @brief Add RobotInfo ordered by id
+        void add_robot(RobotInfo new_robot);
+
+        /// @brief Return the index of the striker
+        uint8_t find_striker();
+
+        /// @brief Decide the correct soccer position
+        void give_directions();
+
+        /// @brief emit purpose based on leader's instructions
+        void follow_directions(const RoboCup& robocup, const uint8_t incoming_robot_id);
+
+        /// @brief leader determines its own soccer position
+        void self_direct(uint8_t striker_idx);
+
+        /// @brief Store robots that can currently play
+        std::vector<RobotInfo> active_robots;
```

Comment:
```suggestion
        std::vector<RobotInfo> active_robots{};
```
initialises it empty